tests/unit/poller/controller/sources/test_devices.py

```diff
@@ -80,7 +134,7 @@ async def test_devices_set(inventory: Dict):
 
         # emulate what the function Source.set_device should do
         exp_inv = {}
-        for key, node in inventory.items():
+        for key, node in result_inventory.items():
```

Comment:
Your fix is absolutely correct, but now you don't need to compare one element at a time. You can simply compare `inv` with `result_inventory`
```suggestion
assert inv == result_inventory, 'inventories do not match'
```